examples/Playground.hs

```diff
@@ -58,6 +64,7 @@ testT = Test
     <*> Toml.table insideCodec "testX" .= testX
     <*> Toml.dioptional ((Toml.table insideCodec) "testY") .= testY
     <*> Toml.diwrap (Toml.text "testN") .= testN
+    <*> Toml.sumType "testC" .= testC
```

Comment:
Technically speaking, `tomland` already has a way to write bidirectional serializers for enum types. You could just use [`Toml.read`](http://hackage.haskell.org/package/tomland-1.0.1.0/docs/Toml-Bi-Combinators.html#v:read). But I see that your `sumType` provides much better error messages, and it's always good to have better error messages!